src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,89 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  originTodos: Todo[];
+  setTodoList: (value: Todo[]) => void;
+}
+
+const prepareTodoList = (
+  originTodos: Todo[],
+  filterParam: string,
+  query: string,
+) => {
+  let copy = [...originTodos];
+
+  copy = copy.filter(({ completed }) => {
+    switch (filterParam) {
+      case 'active':
+        return !completed;
+      case 'completed':
+        return completed;
+      default:
+        return true;
+    }
+  });
+
+  if (query) {
+    const normalizedQuery = query.trim().toLowerCase();
+
+    copy = copy.filter(({ title }) =>
+      title.toLowerCase().includes(normalizedQuery),
+    );
+  }
+
+  return copy;
+};
+
+export const TodoFilter: React.FC<Props> = ({ originTodos, setTodoList }) => {
+  const [query, setQuery] = useState('');
+  const [filterParam, setFilterParam] = useState('all');
+
+  useEffect(() => {
+    setTodoList(prepareTodoList(originTodos, filterParam, query));
+  }, [originTodos, query, filterParam, setTodoList]);
```

Comment:
```suggestion
  const preparedTodos = prepareTodoList(originTodos, filterParam, query);
```
Don't set them into a state, just store in a variable